pymodbus/server/async_io.py

```diff
@@ -577,11 +567,7 @@ async def async_stop(cls):
         """Wait for server stop."""
         if not cls.active_server:
             raise RuntimeError("ServerAsyncStop called without server task active.")
-        await cls.active_server.server.shutdown()
-        if os.name == "nt":
-            await asyncio.sleep(1)
-        else:
-            await asyncio.sleep(0)
+        await cls.active_server.shutdown()
```

Comment:
active_server is a _serverlist object, not the real server (name is quite confusing though).

Reply:
This comment and below "These are needed to allow the async task to run" got confused because the diff view does not show the changes well.

Really the changes are this:
```python
-- await cls.active_server.server.shutdown()
++ await cls.active_server.shutdown()
```
The new code shuts down a server correctly since `active_server` is now a `server` object.

```
await cls.active_server.shutdown()
-- if os.name == "nt":
--  await asyncio.sleep(1)
-- else:
--  await asyncio.sleep(0)
```
I do not see why the sleeps are "needed to allow the asyncio tasks to run".  The `await cls.active_server.shutdown()` already yields control flow back to the event loop and the `await asyncio.sleep()` is redundant, no?

Reply:
We have had severe problems in the past, but not forcing a task switch which is essentially what asyncio.sleep(0) does. Not having a task switch mean e.g. that the task is cancelled but not yet terminated, not always but sometimes (especially on windows).

Reply:
So you are saying there are no reason to keep a list of active servers, but then we could remove _server_list totally.

_server_list was build to maintain a list of active server, which comes handy when debugging...I do not like to keep _server_list without the list.